build/Platform.targets

```diff
@@ -0,0 +1,156 @@
+
+    <!-- Clear working directory -->
+    <RemoveDir Directories="$(_WorkRoot)" />
+
+    <PropertyGroup>
+      <PlatformWorkDirectory>$(_WorkRoot)Platform\</PlatformWorkDirectory>
+      <PlatformOutputDirectory>$(_WorkRoot)Publish\</PlatformOutputDirectory>
+      <PlatformPublishDirectory>$(PlatformOutputDirectory)shared\Microsoft.AspNetCore.All\$(PackageVersion)\</PlatformPublishDirectory>
+
+      <_PlatformProps>Configuration=$(Configuration);RuntimeIdentifier=$(PlatformRID)</_PlatformProps>
+      <_PlatformProps>$(_PlatformProps);RuntimeFrameworkVersion=$(MicrosoftNETCoreApp20PackageVersion)</_PlatformProps>
+      <_PlatformProps>$(_PlatformProps);DotNetRestoreSourcePropsPath=$(GeneratedRestoreSourcesPropsPath)</_PlatformProps>
+      <_PlatformProps>$(_PlatformProps);AspNetUniverseBuildOffline=true</_PlatformProps>
+    </PropertyGroup>
+
+    <!-- Move to working dir -->
+    <ItemGroup>
+      <PlatformFiles Include="$(_TemplatesDir)Platform\**\*" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(PlatformFiles)" DestinationFolder="$(PlatformWorkDirectory)\%(RecursiveDir)" />
+    <Copy SourceFiles="$(_SrcDir)Directory.Build.props" DestinationFolder="$(_WorkRoot)" />
+
+    <!-- Set _Target=Restore so the project will be re-evaluated to include Internal.AspNetCore.Sdk MSBuild properties on the next step. -->
+    <MSBuild Projects="$(PlatformWorkDirectory)Platform.csproj"
+      Targets="Restore"
+      Properties="$(_PlatformProps);MicrosoftAspNetCoreAllVersion=$(PackageVersion);_Target=Restore" />
+
+    <!-- Publish -->
+    <MSBuild Projects="$(PlatformWorkDirectory)Platform.csproj"
+      Targets="Publish"
+      Properties="$(_PlatformProps);GenerateRuntimeConfigurationFiles=true;SelfContained=false;PublishDir=$(PlatformPublishDirectory)" />
+
+    <!-- Clean deps.json -->
+    <RepoTasks.TrimDeps DepsFiles="$(PlatformPublishDirectory)/Platform.deps.json" />
+
+    <!-- Clean up artifacts that publish generates which we don't need -->
+    <ItemGroup>
+      <ToDelete Include="$(PlatformPublishDirectory)\Platform" />
+      <ToDelete Include="$(PlatformPublishDirectory)\Platform.dll" />
+      <ToDelete Include="$(PlatformPublishDirectory)\Platform.pdb" />
+    </ItemGroup>
+
+    <Delete Files="@(ToDelete)" />
+
+    <!-- Rename deps file -->
+    <Move SourceFiles="$(PlatformPublishDirectory)\Platform.deps.json"
+          DestinationFiles="$(PlatformPublishDirectory)\Microsoft.AspNetCore.All.deps.json" />
+
+    <!-- Rename runtimeconfig.json file -->
+    <Move SourceFiles="$(PlatformPublishDirectory)\Platform.runtimeconfig.json"
+          DestinationFiles="$(PlatformPublishDirectory)\Microsoft.AspNetCore.All.runtimeconfig.json" />
+
+    <!-- Generate Runtime Graph -->
+    <PropertyGroup>
+      <RuntimeGraphGeneratorRuntime Condition="$([MSBuild]::IsOSPlatform('Windows'))">win</RuntimeGraphGeneratorRuntime>
```

Comment:
Other than crossgen, is there any reason I can't build and gather the Linux and macOS shared platforms on a Windows machine?

Reply:
Again, trying to use the same logic as core-setup. I don't have a clear understanding of the underlying mechanism. I can diff the outputs to ensure this doesn't break but I'd like to do that after we produce the first version.